src/inmanta/env.py

```diff
@@ -378,19 +378,23 @@ def _run_pip_install_command(
                 constraints_files=constraints_files,
                 requirements_files=requirements_files,
             )
-            self._run_command_and_log_output(cmd, stderr=subprocess.PIPE)
-        except CalledProcessError as e:
-            stderr: str = e.stderr.decode()
-            not_found: List[str] = [
-                requirement.project_name
-                for requirement in requirements
-                if f"No matching distribution found for {requirement.project_name}" in stderr
-            ]
-            if not_found:
-                raise PackageNotFound("Packages %s were not found in the given indexes." % ", ".join(not_found))
-            if "versions have conflicting dependencies" in stderr:
-                raise ConflictingRequirements(stderr)
-            raise e
+            return_code, full_output = self._run_command_and_stream_output(cmd)
+
+            if return_code:
+                not_found: List[str] = []
+                conflicts: bool = False
+                for line in full_output:
+                    if "No matching distribution found for " in line:
+                        # Add missing package name to not_found list
```

Comment:
As far as I can see we don't have a test case to verify the log lines when no matching distribution if found. I think we need to add one.

Reply:
Added test : `test_no_matching_distribution`